examples/hello_compute_rust/main.rs

```diff
@@ -0,0 +1,107 @@
+extern crate env_logger;
+extern crate wgpu;
+extern crate wgpu_native;
+
+use std::str::FromStr;
+
+// TODO: deduplicate this with the copy in gfx-examples/framework
+pub fn cast_slice<T>(data: &[T]) -> &[u8] {
+    use std::mem::size_of;
+    use std::slice::from_raw_parts;
+
+    unsafe { from_raw_parts(data.as_ptr() as *const u8, data.len() * size_of::<T>()) }
+}
+
+fn main() {
+    env_logger::init();
+
+    // For now this just panics if you didn't pass numbers. Could add proper error handling.
+    if std::env::args().len() == 1 {
+        panic!("You must pass a list of positive integers!")
+    }
+    let numbers: Vec<u32> = std::env::args()
+        .skip(1)
+        .map(|s| u32::from_str(&s).expect("You must pass a list of positive integers!"))
+        .collect();
+
+    let size = (numbers.len() * std::mem::size_of::<u32>()) as u32;
+
+    let instance = wgpu::Instance::new();
+    let adapter = instance.get_adapter(&wgpu::AdapterDescriptor {
+        power_preference: wgpu::PowerPreference::LowPower,
+    });
+    let mut device = adapter.create_device(&wgpu::DeviceDescriptor {
+        extensions: wgpu::Extensions {
+            anisotropic_filtering: false,
+        },
+    });
+
+    let cs_bytes = include_bytes!("./../data/collatz.comp.spv");
+    let cs_module = device.create_shader_module(cs_bytes);
+
+    let staging_buffer = device.create_buffer(&wgpu::BufferDescriptor {
+        size,
+        usage: wgpu::BufferUsageFlags::MAP_READ
+            | wgpu::BufferUsageFlags::TRANSFER_DST
+            | wgpu::BufferUsageFlags::TRANSFER_SRC,
+    });
+    staging_buffer.set_sub_data(0, cast_slice(&numbers));
+
+    let storage_buffer = device.create_buffer(&wgpu::BufferDescriptor {
+        size: (numbers.len() * std::mem::size_of::<u32>()) as u32,
+        usage: wgpu::BufferUsageFlags::STORAGE
+            | wgpu::BufferUsageFlags::TRANSFER_DST
+            | wgpu::BufferUsageFlags::TRANSFER_SRC,
```

Comment:
why does it need `TRANSFER_SRC`?

Reply:
I had assumed I'd need that to pass validation when I copy the storage buffer back to the staging buffer?

Reply:
I missed the fact you are copying back to it.